### PR TITLE
Issue 4868: Enable mkdocs plugin in build scripts 

### LINKED ISSUE #4868: Enable mkdocs plugin in gradle build scripts
**Problem description**
Currently in order to generate a documentation for pravega.io website one needs to install all necessary dependencies in order to use `mkdocs` tool. This a bit odd and there should be a better way to handle this.  

**Problem location**
Build scripts 

**Suggestions for an improvement**
Enable `mkdocs` plugin in gradle so it will take care of pulling all the dependencies

### PR DESCRIPTION
**Pull request by medvedevigorek**
**Change log description**  
Enables `mkdocs` gradle plugin to make it easier to generated the documentation as well as modify it. In addition also fixes most of the warnings by adding missing pages or fixing the broken links. 

**Purpose of the change**  
 Closes #4868 

**What the code does**  

1. Enables `mkdocs` plugin in `gradle.properties` and also bumps its version to 1.1 
2. Replaces deprecated `pages` configuration with `nav` 
3. Fixes broken link to `README` file in `Getting Started` page
4. Adds `Securing Pravega in Standalone Mode` under `Pravega Security` as the page was in there but never made available
5. Updates a link to Client javadoc location as it gets corrected every time when a new version of documentation is published.  
6. Removes configuration related to `publish` task as auto-publishing is not yet fully supported

**How to verify it**  
Manually verified that `mkdocs` tasks are now available as gradle tasks and that both `mkdocsBuild` as well as `mkdocsServe` work in both Windows and Linux environments.

### CODE REVIEW

**Comment by shrids** [comment]
The changes look good on inspection; Couple of observations
- The latest plugin version is 2.0.1 (https://github.com/xvik/gradle-mkdocs-plugin/tags) We are not using it since it does not work with gradle 4.x
- I tried `gradle mkdocsBuild` with this changeset and it failed on my Linux box. (perhaps it is a setup issue on my laptop)
```
[python] /usr/bin/python3.6 -m mkdocs build -c -d /home/osboxes/code/pravega/pravega/documentation/generated/0.8.0-2582.f550e4a-SNAPSHOT
         /usr/bin/python3.6: No module named mkdocs
FAILURE: Build failed with an exception.
```

**Comment by medvedevigorek** [comment]
That's a good catch @shrids! While it works on both of my machines (windows and linux) I gave it another try on a fresh linux box (deployed an ubuntu container and installed all missing dependencies including python+pip) and observed the same issue. 

What actually happens is `gradle-mkdocs-plugin` depends on `ru.vyarus:gradle-use-python-plugin` plugin and the latter works in getting `virtualenv` prepared but fails to execute other tasks in the created virtualenv. The actual fix was in both of them and the latest version of both should work fine however since we are locked with gradle 4.x there is no way we can upgrade them. at least not now. 

It used to work for me before because I installed all necessary dependencies in the system pythong env hence everything works when plugin invoked system python. 

I've been trying various options on getting `mkdocsBuild` tasks executed in the created virtualenv however so far no luck. Currently I am getting more and more towards idea to let the plugin to install all the dependencies in the user python installation and keep the idea of using the dedicated `virtualenv` till better times. 

Will update PRs in coming day once test it be a bit better.

**Comment by medvedevigorek** [comment]
@shrids I've finally made the changes, there will be no virtualenv used at all, instead all the dependencies are installed in the user python directory.

File: gradle.properties [hunk 1]
```diff
@@ -66,3 +66,6 @@ bookkeeperBaseTag=pravega/bookkeeper
 signing.keyId=
 # This will be defaulted to ~/.gnupg/secring.gpg if not provided as a command line property
 signing.secretKeyRingFile=
+
+enableMkdocs=true
```
Comment (shrids) [inline gradle.properties:70]: Is there a reason why it was disabled in the older build?

Comment (medvedevigorek) [reply gradle.properties:70]: It was enabled by default initially (see PR #2169) but then your question made me checking the history of some other files and it seems there was an issue with building Flink Connector (see Issue #2423).  @crazyzhou  is helping to check whether the issue is still present if `mkdocs`  plugin is enabled by default or not.

**Review by andreipaduroiu — DISMISSED** [review]
At the request of @medvedevigorek , blocking this until we resolve downstream issues with the connectors.

**Review by derekm — APPROVED** [review]
I think this is read to merge. This PR is needed to collaborate on docs tech debt items.

**Review by RaulGracia — APPROVED** [review]

**Review by shrids — APPROVED** [review]

**Comment by RaulGracia** [comment]
Checked with @crazyzhou that Flink connector removed the Pravega submodule in PR https://github.com/pravega/flink-connectors/pull/470.

**Pull request merged**